src/main/java/bisq/core/dao/node/consensus/OpReturnVoteController.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * This file is part of Bisq.
+ *
+ * Bisq is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or (at
+ * your option) any later version.
+ *
+ * Bisq is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public
+ * License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with Bisq. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package bisq.core.dao.node.consensus;
+
+import bisq.core.dao.blockchain.ReadableBsqBlockChain;
+import bisq.core.dao.blockchain.vo.Tx;
+import bisq.core.dao.blockchain.vo.TxOutput;
+import bisq.core.dao.blockchain.vo.TxOutputType;
+import bisq.core.dao.blockchain.vo.TxType;
+
+import bisq.common.app.Version;
+
+import javax.inject.Inject;
+
+import lombok.extern.slf4j.Slf4j;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+@Slf4j
+public class OpReturnVoteController {
```

Comment:
In relation to my last comment regarding DDD naming conventions and semantics, I notice that there are a number of `Controller` classes within the `dao.node.consensus` package. We haven't used `Controller` naming anywhere else in the Bisq codebase except bisq-pricenode, where I'm using it in the conventional DDD sense, and indeed using Spring's `@Controller` annotations there.

Is there a particular reason why `Controller` is the term of choice here? Since these aren't "web controllers", and that's what `Controller` has come to be synonymous with in the DDD / modern Java application development world, is there a better naming we could go with instead?

I've just taken a cursory glance through all the types in that package, and one common thread among them looks to be validation / verification, which makes sense given that this package is about consensus rules, and it looks like some types also "apply state changes". I'd suggest refactoring toward something like `Validator` and `Mutator` naming, and avoid overloading the well-known semantics of DDD `Controller`, `Service`, etc naming.

Reply:
Yes they do usually both validation and applying a state change. I did not wanted to split them as that seems too much to have for each method a class... The reason for controller came for me from the classical MVC where the controller is taking input data, verifies it and applies on the model state change. 
Might be not good as well as it's not a UI part... but did not find a better name yet.